### PR TITLE
Add minLength and checkComplexity password policy functionality

### PR DESCRIPTION
**Pull request by stephdl**
This pull request adds the functionality 

to enforce a minimum length for passwords in the ChangeUserPasswordModal and CreateOrEditUserModal components. The minimum length is set to 8 characters.

to check the complexity of the password (upper, lower, digit and symbol)
  
- strength disabled
image
- strength enabled but complexity disabled
image
- strength enabled and complexity enabled
image

- validation 
Capture d’écran du 2024-02-20 14-50-55
Capture d’écran du 2024-02-20 14-50-14
Capture d’écran du 2024-02-20 14-49-15

https://github.com/NethServer/dev/issues/6845

### CODE REVIEW

File: core/ui/src/components/domains/ChangeUserPasswordModal.vue [hunk 1]
```diff
@@ -90,14 +112,51 @@ export default {
...
+      const err = res[0];
+
+      if (err) {
+        console.error(`error creating task ${taskAction}`, err);
+        this.error.listPasswordPolicy = this.getErrorMessage(err);
+        return;
+      }
+    },
+    listPasswordPolicyCompleted(taskContext, taskResult) {
+      const Config = taskResult.output;
```
Comment (andre8244) [inline core/ui/src/components/domains/ChangeUserPasswordModal.vue:152]: Let's declare variables with leading lowercase letter

File: core/ui/src/components/domains/CreateOrEditUserModal.vue [hunk 2]
```diff
@@ -209,6 +232,43 @@ export default {
...
+      const err = res[0];
+
+      if (err) {
+        console.error(`error creating task ${taskAction}`, err);
+        this.error.listPasswordPolicy = this.getErrorMessage(err);
+        return;
+      }
+    },
+    listPasswordPolicyCompleted(taskContext, taskResult) {
+      const Config = taskResult.output;
```
Comment (andre8244) [inline core/ui/src/components/domains/CreateOrEditUserModal.vue:264]: Let's declare variables with leading lowercase letter

Comment (stephdl) [reply core/ui/src/components/domains/CreateOrEditUserModal.vue]: should be good

File: core/ui/src/components/domains/ChangeUserPasswordModal.vue (hunk 1 above)
Comment (stephdl) [reply core/ui/src/components/domains/ChangeUserPasswordModal.vue]: should be good

**Review by andre8244 — APPROVED** [review]
LGTM, I suggest a squash & merge

**Pull request merged**